docs/dinamicko-programiranje/najdulji-rastuci-podniz.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Najdulij rastući podniz
+---
+
+### Problem
+Jedan od klasičnih problema dinamičkog programiranja je određivanje najduljeg rasutećeg podniza zadanog niza.
+
+Zadan je niz duljine $N$ čiji su elementi $x_i < 1e5$. Odredi najdulji podniz, ne nužno uzastopnih elemenata, za čije elemente vrijedi da je svaki element veći ili jednak od elemenata prije njega.
+
+### Rješenje
+
+Za ovaj problem postoji jednostavno rješenje. Neka nam je stanje najdulji podniz do trenutnog elementa, a funkcija prijelaza je prolazak po svim prethodnim elementima gdje od svih elemenata koji su manji ili jednaki trenutnom tražimo onaj koji do tada tvori najdulji podniz, odnosno onaj u čijoj je dinamici zapisan najveći broj. Tom broju dodajemo 1 te to postaje vrijednost trenutnog stanja dinamike.
```

Comment:
najdulji podniz razmatrajući niz do trenutnog elementa, koji uključuje taj element. (ili završava njim, vidi što bolje zvuči)

Reply:
Neka nam je stanje najdulji podniz razmatrajući niz do nekog elementa(uključujući taj element)

Jel ok

Reply:
duljina podniza